spec/setupTests.ts

```diff
@@ -17,3 +17,9 @@ limitations under the License.
 import DOMException from "domexception";
 
 global.DOMException = DOMException;
+
+jest.mock("../src/http-api/utils", () => ({
+    ...jest.requireActual("../src/http-api/utils"),
+    // We mock timeoutSignal otherwise it causes tests to leave timers running
+    timeoutSignal: () => new AbortController().signal,
+}));
```

Comment:
I'm getting pretty lost here. This stuff doesn't really seem to be related the stated purpose of this PR (running the tests in multiple node versions): is it about making it possible to run the suites separately?  My instinct is to say that it should be pulled out to a separate PR.

Alternatively: can you explain what this is doing, and why it is needed now where it wasn't before?

Reply:
It is for Node 19 compatibility, where things are stricter

Reply:
so are all the changes to the `spec` files node 19 compat? or are there some changes for other things?

it really feels like this PR is conflating 3 things and I would love to see it split up

Reply:
Node 19 compat, tests were hanging and Jest was failing them due to timers being left hanging